### ISSUE TITLE
Log the number of consumed operations per issue

### ISSUE THREAD

**Issue by C0ZEN** [actions/stale#348]
Since the option `operations-per-run` is very hard to understand and to track, adding a log each time an issue is processed indicating the quantity of consumed operations could be a nice thing to have.
This log would appear only if there is some consumed.

Also it should indicates the left count.

**Comment by lucacome** [comment]
I have this to the default value and I keep getting `Reached max number of operations to process. Exiting.`, but I'm not sure what raising the limit above 30 would entail. Do you know? Maybe some documentation around this would be good.

**Comment by C0ZEN** [comment]
@lucacome 
The stale action will stop sooner than expected and will process less issues and PRs so no risk at all for this stale action.
The only drawback is that if you have some other actions using the same "user" than the one used for the stale action then those other actions will no longer be able to call the GitHub API until GitHub reset the counter (I think they give you something like 5K API calls per hour at least).
It's basically a security to avoid flooding.

I suggest you to increase the limit to 100 and see if the warning message is still there.
Basically, depending on the configuration of the stale action and the quantity of issues and PRs the action that consume the number of operations can be very different.

Reading once again the doc https://github.com/actions/stale#operations-per-run about this option made me think that it's not clear enough what is causing the consumption of this limit and what are the consequences when reaching them.
I will update it then.

**Comment by C0ZEN** [comment]
@lucacome FYI I updated the doc (in a PR) to this:

#### operations-per-run

_Context:_  
The action will perform some API calls to GitHub to fetch or close issues and pull requests, set or update labels, add comments, delete branches, etc.  
These operations are made in a very short period of time - because the action is very fast to run - and can be numerous based on your project action configuration and the quantity of issues and pull requests within it.  
GitHub has a [rate limit](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting) and if reached will block all of these API calls for one hour (or API calls from other actions using the same user (a.k: the github-token from the [repo-token](#repo-token) option)).  
This option aim to avoid crossing this situation by limiting the number of operations for a single run.

_Purpose:_  
This option aim to limit the number of operations made with the GitHub API to avoid reaching the [rate limit](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting).

Based on your project, your GitHub business plan and the date of the cron job you set for this action, you can increase this limit to a higher number.
If you are not sure which is the right value for you or if the default value is good enough, you could enable the logs and look at the end of the stale action.  
If you reached the limit, a warning will be visible and will suggest you to increase the number of operations.  
If you don't increase it, the risk is to have un-processed issues or pull requests, that's all.

When [debugging](#Debugging), you can set it to a much higher number like `1000` since there will be fewer operations made with the GitHub API.  
Only the [actor](#repo-token) and the batch of issues (100 per batch) will consume the operations.

Default value: `30`

**Comment by lucacome** [comment]
Thanks @C0ZEN this helps!

**Comment by C0ZEN** [comment]
up
